patches/components-component_updater-component_updater_service.cc.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/components/component_updater/component_updater_service.cc b/components/component_updater/component_updater_service.cc
+index 49ce676e11a53eaae623b874f0d1b23415bc93a9..545b5067a2a225d43fa8e46ef506f6fba88e07f1 100644
+--- a/components/component_updater/component_updater_service.cc
++++ b/components/component_updater/component_updater_service.cc
+@@ -475,8 +475,8 @@ std::unique_ptr<ComponentUpdateService> ComponentUpdateServiceFactory(
+   DCHECK(config);
+   DCHECK(scheduler);
+   auto update_client = update_client::UpdateClientFactory(config);
+-  return std::make_unique<CrxUpdateService>(config, std::move(scheduler),
```

Comment:
You should be able to do this without patches, see code in: `chrome/browser/extensions/extension_install_prompt.cc`

Reply:
Thank you very much for the suggestion. This technique was new to me, and it seems very elegant. Unfortunately, after trying to use it here (see my commits, esp. `Use #define to avoid having to create patches`), I'm not sure it will be feasible.

In `extension_install_prompt`, which you mentioned, the overwritten method `GetDialogTitle` is public:

    class Prompt {
     public:
      ...
      base::string16 GetDialogTitle() const;

Furthermore, and importantly, `GetDialogTitle` is only called from _outside_ `extension_install_prompt.cc`. This means that the macro that renames `GetDialogTitle` to `GetDialogTitle_ChromiumImpl` has no effect on other files. That is, other source files will call the _overwritten_ `GetDialogTitle` and not the original `GetDialogTitle_ChromiumImpl`.

In the present case of `component_updater_service.cc`, the callers of the overwritten method `CheckForUpdates` are unfortunately in the same file. That is, when I define a macro that renames it to `CheckForUpdates_ChromiumImpl`, then the call sites in that file are _also_ renamed. So even if I then define a new implementation of `CheckForUpdates`, that implementation won't be called because `CheckForUpdates_ChromiumImpl` is called instead.

I could fix this by copying method definitions until I get to the public entry point in this situation, which is the method `RegisterComponent`. Precisely: The public method `RegisterComponent` calls the private method `Start`, which in turn calls the private method `CheckForUpdates`:

    CrxUpdateService::RegisterComponent(...) {
        ...
        Start();
        ...
    }

    CrxUpdateService::Start(...) {
        ...
        CheckForUpdates();
        ...
    }

These are all in `component_updater_service.cc`, so the `CheckForUpdates()` call gets replaced by `CheckForUpdates_ChromiumImpl()`. What I would have to do is copy their implementations into `brave-core`. But that's quite a lot of code and very likely to break or become stale as Chromium is updated.

Can you advise what you think I should do @jumde?

Reply:
@mherrmann - What do you think about creating a chromium_src override for `UpdateClientImpl::Update` instead?

Reply:
@jumde I don't think it will work in `UpdateClientImpl::Update` because `CrxDataCallback` is a `OnceCallback`. (The obvious way to implement your suggestion would be to have a `for` loop in `UpdateClientImpl::Update` that submits multiple update tasks instead of a single one. But because of the `OnceCallback`, only the first of these tasks would be able to query the current components.)

After some more analysis, I think the best place is in `update_checker`, injected via `UpdateClientFactory`. I've committed an implementation of this. If you could be so kind to review this, that would be great. From my side, I would deem it merge-able. Before doing so, I would squash my commits and (if you think I should) remove the logging calls. Please let me know :-)